backend/pkg/backend/backend.go

```diff
@@ -83,6 +84,12 @@ func Run() {
 	dbConfig := createDatabaseConfig(config)
 	dbHandler := _database.Init(dbConfig)
 
+	dbHandler.SetMaterializedViewHandler(dbConfig)
+	if config.DatabaseDriver == _database.DBDriverTypePostgres {
+		dbHandler.ViewRefreshHandler.RegisterViewRefreshHandler(_database.RefreshMaterializedViews)
```

Comment:
I think perhaps you misunderstood the intension of RegistryViewRefreshHandler in my previous comment, the intension what that inside the of the createAllViews function you would use it to register a refresh handler for each view for example:
```
func createAllViews(db *gorm.DB, dbDriver string) {
        ....

	if err := db.Exec(fmt.Sprintf(applicationsViewQuery, createCommand)).Error; err != nil {
		log.Fatalf("Failed to create applications_view: %v", err)
	}
	if dbDriver == DBDriverTypePostgres {
	       // Register postgres view refresh *if* applications, resources or packages tables change
	       // those are the only tables that view requires.
	       dbHandler.ViewRefreshHandler.RegisterViewRefreshHandler(
	               func(db *gorm.DB){
	                       if err := db.Exec(fmt.Sprintf(refreshMaterializedViewConcurrentlyCommand, applicationsViewName)).Error; err != nil {
	                               log.Errorf("Failed to refresh materialized %s: %v", applicationsViewName, err)
		               }
                       },
                       "applications",
                       "resources",
                       "packages",
               )
	}
	
	...
	
        if err := db.Exec(fmt.Sprintf(vulnerabilitiesViewQuery, createCommand)).Error; err != nil {
		log.Fatalf("Failed to create vulnerabilities_view: %v", err)
	}
	if dbDriver == DBDriverTypePostgres {
	       // Register postgres view refresh *if* vulnerability, resources or packages tables change
	       dbHandler.ViewRefreshHandler.RegisterViewRefreshHandler(
	               func(db *gorm.DB){
	                       if err := db.Exec(fmt.Sprintf(refreshMaterializedViewConcurrentlyCommand, vulnerabilityViewName)).Error; err != nil {
	                               log.Errorf("Failed to refresh materialized %s: %v", vulnerabilityViewName, err)
		               }
                       },
                       "vulnerabilities",
                       "resources",
                       "packages",
               )
        }
	...
}
```

Reply:
I understood what you wanted in the previous review, but I think we don't need to register a handler for all views because we want to refresh only one way in case of a specified dbDriver. So we need to register the handler for a dbDriver and do the refresh iterating over the required views.

Reply:
that won't be as generic and flexible as registering a refresh handler per view, it might satisfy our current use case but it doesn't leave much room for expansion/extension in the future if we need to do more things for one view over a different view. If you want to reduce duplication but keep the flexibility then you can just do:

```
func createAllViews() {
       ...
       dbHandler.ViewRefreshHandler.RegisterViewRefreshHandler(
            getPostgresViewRefreshHandlerFunc(vulnerabilityViewName),
            "table1",
            "table2",
            "table3",
            "table4",
       )
       ...
}


func getPostgresViewRefreshHandlerFunc(viewname string) func(db *gorm.DB) {
     return func(db *gorm.DB) {
             if err := db.Exec(fmt.Sprintf(refreshMaterializedViewConcurrentlyCommand, viewname)).Error; err != nil {
	             log.Errorf("Failed to refresh materialized %s: %v", viewname, err)
             }
     }
}
```

this still leaves the option for a view that needs to it have its own handler that does other things.